WordPress/Classes/ViewRelated/NUX/SigninEmailViewController.swift

```diff
@@ -286,10 +286,30 @@ import WordPressShared
             return
         }
 
+        let service = AccountService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+
         guard emailOrUsername.isValidEmail() else {
             // A username was entered, not an email address.
             // Proceed to the next form:
-            if !SigninHelpers.isUsernameReserved(emailOrUsername) {
+            if SigninHelpers.isWPComDomain(emailOrUsername) {
```

Comment:
With the addition of the new logic this guard block is getting a bit unwieldy. Might be a good opportunity to refactor as well.